gunicorn/arbiter.py

```diff
@@ -573,7 +535,12 @@ def manage_workers(self):
         workers = sorted(workers, key=lambda w: w[1].age)
         while len(workers) > self.num_workers:
             (pid, _) = workers.pop(0)
-            self.kill_worker(pid, signal.SIGTERM)
+            worker = self.WORKERS[pid]
+            if worker.killed['by'] is None:
+                self.kill_worker(pid, signal.SIGTERM)
+            elif (worker.killed['by'] == signal.SIGTERM
+                  and time.monotonic() > worker.killed['when'] + worker.timeout):
+                self.kill_worker(pid, signal.SIGKILL)
```

Comment:
Yes, looking at it again, it is true that the USR1 handling didn't fit into the suggestion! Once and if this PR lands, I could look into posting a follow-up PR.

Resolving since the patch is removed from the patch set.

Reply:
I tried a slightly different approach at implementing this here: https://github.com/sylt/gunicorn/pull/1. Hopefully, the USR1 problem should be solved!

Even though the new approach applies cleanly on master, it depends on work done in this PR. So, I'll take it one thing at a time.